flask/app/analyses.py

```diff
@@ -173,11 +174,37 @@ def create_analysis():
     requested = now
     updated = now
     analysis_state = "Requested"
+
+    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
+        user_id = request.args.get("user")
+    else:
+        user_id = current_user.user_id
+
     try:
-        requester_id = updated_by_id = current_user.user_id
+        requester_id = updated_by_id = user_id
     except:  # LOGIN DISABLED
         requester_id = updated_by_id = 1
```

Comment:
This should no longer be a try-except and just a simple assignment that falls back to `current_user.user_id`

Reply:
this will now fail if there is no user_id supplied